File: 56. Merge Intervals - leetcode /56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
         vector<pair<int, int>> arr;

        for (int i = 0; i < intervals.size(); i++) {
            arr.push_back(make_pair(intervals[i][0], intervals[i][1]));
        }

        sort(arr.begin(), arr.end());

        int res = 0;

        for (int i = 1; i < intervals.size(); i++) {
            if (arr[i].first <= arr[res].second) {
                arr[res].second = max(arr[i].second, arr[res].second);
                arr[res].first = min(arr[i].first, arr[res].first);
            } else {
                res++;
                arr[res] = arr[i];
            }
        }

        vector<vector<int>> result(res + 1, vector<int>(2, 0));

        for (int i = 0; i <= res; i++) {
            result[i][0] = arr[i].first;
            result[i][1] = arr[i].second;
        }

        return result;
    }
};

File: 56. Merge Intervals - leetcode /56. Merge Intervals.java
class Pair implements Comparable<Pair>
{ 
    int s, e; 
    Pair(int s, int e){
        this.s = s;
        this.e = e;
    }
    public int compareTo(Pair a)  
    { return this.s - a.s; }  // sort accordin to start time
};

class Solution
{
    public int[][] merge(int[][] Intervals)
    {
        // Code here // Code here
      Pair  arr[] = new Pair[Intervals.length];
    //   int arr[]= new int[]
    

    for( int i =0; i<Intervals.length; i++)
        arr[i]  = new Pair(Intervals[i][0], Intervals[i][1]);

    Arrays.sort(arr);
    int res = 0;
    
    for( int i =1; i<Intervals.length; i++){
        if(arr[i].s <= arr[res].e){
            arr[res].e = Math.max(arr[i].e, arr[res].e);
            arr[res].s = Math.min(arr[i].s , arr[res].s);
        }
        else {
            res++; arr[res]=arr[i];
        }
    }
    int al[][] = new int[res+1][2];
        for(int i =0; i<=res;i++){
            al[i][0]= arr[i].s;
            al[i][1] = arr[i].e;
        }

    return al;
    }
}
